src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/clusterinteraction/ClusterDetailFragment.kt

```diff
@@ -182,37 +181,55 @@ class ClusterDetailFragment : Fragment() {
   ) {
     responseValues.forEach { (variableNameType, response) ->
       if (response is List<*>) {
-        if (response.size == 0) {
-          val emptyCallback =
-            inflater.inflate(R.layout.cluster_callback_item, null, false) as ConstraintLayout
-          emptyCallback.clusterCallbackNameTv.text = "Result is empty"
-          callbackList.addView(emptyCallback)
-        } else {
-          response.forEachIndexed { index, it ->
-            val objectCallback =
-              inflater.inflate(R.layout.cluster_callback_item, null, false) as ConstraintLayout
-            objectCallback.clusterCallbackNameTv.text = variableNameType.name + "[$index]"
-            val objectDeserializationString = it.toString()
-            val callbackClassName = it!!.javaClass.toString().split('$').last()
-            objectCallback.clusterCallbackDataTv.text = callbackClassName
-            objectCallback.clusterCallbackDataTv.setOnClickListener {
-              AlertDialog.Builder(requireContext())
-                .setTitle(callbackClassName)
-                .setMessage(objectDeserializationString)
-                .create()
-                .show()
-            }
-            objectCallback.clusterCallbackTypeTv.text = "List<$callbackClassName>"
-            callbackList.addView(objectCallback)
-          }
-        }
+        createListReadAttributeView(response, inflater, callbackList, variableNameType)
       } else {
-        val callback =
+        createPrimitiveReadAttributeView(response, inflater, callbackList, variableNameType)
+      }
+    }
+  }
+
+  private fun createPrimitiveReadAttributeView(
```

Comment:
This method covers single `String` and `byte[]` response types, which are not primitives. Maybe `createBasicReadAttributeView`?